### ISSUE TITLE
Partial Match happening for Jpeg Files

### ISSUE THREAD

**Issue by saurabheights** [j256/simplemagic#21]
Hi Gray,
 Since upgrading SimpleMagick to 1.10, JPEG images are getting partial matched. Is this change intentional?

Sample Images: 

![1483354121587_91_535afce2-1faf-4f77-bad9-b06e1f7afec0](https://cloud.githubusercontent.com/assets/11634769/21984751/62d9a9e2-dc1d-11e6-9815-1183d1b04478.jpeg)
![zwtc_1483330847796_77fff134_a228](https://cloud.githubusercontent.com/assets/11634769/21984752/62db360e-dc1d-11e6-9d93-1fe0f62a1e35.jpeg)

Also, I noticed common trait in the two image to have TIFF image data

```
$> file a.jpeg 
a.jpeg: JPEG image data, Exif standard: [TIFF image data, big-endian, direntries=6, software=Aviary for Android 4.4.6, orientation=[*0*], model=SM-A500G, datetime=2016:02:20 12:50:44, manufacturer=SAMSUNG], baseline, precision 8, 530x714, frames 3

$> file b.jpeg 
b.jpeg: JPEG image data, Exif standard: [TIFF image data, little-endian, direntries=0], baseline, precision 8, 1800x2400, frames 3

```
P.S. This is the same user as @se7en007.

**Comment by j256** [comment]
Hrm.  Just spent a couple of hours on this.  I've also added your image to test bank.

For now the partial for this image type is correct.  I don't know how to read it.  It doesn't conform to any known patterns.  If you can find a unix `file` command that outputs more than EXIF standard, I'd like to see it.

**Comment by j256** [comment]
Ok I've one of the jpeg types but it looks like the ones you've posted implement a newer "recursive" magic pattern that simplemagic doesn't understand.  I've added it to the tests though and documented in the TODO.

**Comment by saurabheights** [comment]
Hi, give me few minutes to analyze more. For now:-

`$ exiftool 62db360e-dc1d-11e6-9d93-1fe0f62a1e35.jpeg 
ExifTool Version Number         : 9.46
File Name                       : 62db360e-dc1d-11e6-9d93-1fe0f62a1e35.jpeg
Directory                       : .
File Size                       : 207 kB
File Modification Date/Time     : 2017:01:16 18:55:52+05:30
File Access Date/Time           : 2017:03:19 02:58:26+05:30
File Inode Change Date/Time     : 2017:03:19 02:57:39+05:30
File Permissions                : rw-rw-r--
File Type                       : JPEG
MIME Type                       : image/jpeg
Exif Byte Order                 : Big-endian (Motorola, MM)
Software                        : Aviary for Android 4.4.6
Orientation                     : Unknown (0)
Camera Model Name               : SM-A500G
Modify Date                     : 2016:02:20 12:50:44
Make                            : SAMSUNG
JFIF Version                    : 1.01
Resolution Unit                 : None
X Resolution                    : 1
Y Resolution                    : 1
Image Width                     : 530
Image Height                    : 714
Encoding Process                : Baseline DCT, Huffman coding
Bits Per Sample                 : 8
Color Components                : 3
Y Cb Cr Sub Sampling            : YCbCr4:2:0 (2 2)
Image Size                      : 530x714

$ file 62db360e-dc1d-11e6-9d93-1fe0f62a1e35.jpeg 
62db360e-dc1d-11e6-9d93-1fe0f62a1e35.jpeg: JPEG image data, EXIF standard`

**Comment by saurabheights** [comment]
@j256 : will the above comment suffice(comment is updated with xxd and EXIF marker information)?

**Comment by j256** [comment]
Yeah I've been looking at the raw bytes and I've also looked at both pages interesting enough (yay google).  Both the II and MM exif file headers should have a 0x8769 (either big or little endian) and I'm just not seeing that.  This is something different.

In looking at the linux file command code a bit (https://github.com/file/file.git), they now support "recursive" patterns which is news to me.  Looks like the jpeg stuff has been rewritten to support this.  I'm not going to support that anytime soon.  Sorry.
